### DISCUSSION TITLE
Does script onchange_ trigger if data is changed?

### DISCUSSION THREAD

**Discussion by rishid** [twpayne/chezmoi#3248] [Q&A] [answered]
I just started using Chezmoi yesterday so not sure if I am just doing something wrong or this is not possible.

If I reference a variable in defined in `.chezmoidata.yaml` in a `run_onchange_*.sh.tmpl`. If the variable changes should I expect the script to run on next `apply`?

Also related question, looking at Tom's [dotfiles](https://github.com/twpayne/dotfiles/tree/master/home/.chezmoiscripts) as a sample, in the `.chezmoiscripts` he has subdirectory of darwin, windows, linux? Is there some undocumented magic that will cause these to run only on those specific OSs?

**Comment by bradenhilton** [comment, marked as answer]
> If I reference a variable in defined in `.chezmoidata.yaml` in a `run_onchange_*.sh.tmpl`. If the variable changes should I expect the script to run on next `apply`?

Yes.

`run_onchange_00-greet.ps1.tmpl`:

```text
Write-Output 'Hello, {{ .name }}'
```

`.chezmoidata.yaml`:

```yaml
name: user
```

```console
❯ chezmoi apply
Hello, user
```

`.chezmoidata.yaml`:

```yaml
name: rishid
```

```console
❯ chezmoi apply
Hello, rishid
```

> Also related question, looking at Tom's [dotfiles](https://github.com/twpayne/dotfiles/tree/master/home/.chezmoiscripts) as a sample, in the `.chezmoiscripts` he has subdirectory of darwin, windows, linux? Is there some undocumented magic that will cause these to run only on those specific OSs?

There is no magic, the subdirectories are just for better organization. They still need to be added to `.chezmoiignore`:

https://github.com/twpayne/dotfiles/blob/0b983a059ea2c4b4cac6e0f5c2c55b2ba22c969d/home/.chezmoiignore.tmpl#L4-L18

**Reply by rishid** [reply]
Thanks for the example, that helped. I had a mistake in my code

I didn’t catch the ignore behavior, that is neat and will use myself.
